### PR TITLE
[MC-669][MC-695][MC-696] Proofpoint Tap - Fix invalid type in custom output in Fetch Forensics action

### PR DESCRIPTION
**Pull request by dsliwinski-r7**
### Description

Describe the proposed changes:

  - Fix invalid type for `blacklisted` in `evidence_type` custom output
  - Add a conversion to a boolean for `blacklisted` returned as an integer in Fetch Forensics action
  - Update unit test for Fetch Forensics action
  - Add unit tests for for all other actions

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

[assessments.txt](https://github.com/rapid7/insightconnect-plugins/files/7076158/assessments.txt)

<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator CloudReadyValidator
[*] Executing validator UnapprovedKeywordsValidator
[*] Executing validator HelpExampleValidator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
violation: help.md, line: 718
violation: help.md, line: 724
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
WARNING: URLs found that return a 4xx code. Verify they are publicly accessible and if not, update with a working URL.
violation: plugin.spec.yaml[838]: https://threatinsight.proofpoint.com/5d1ce8c6-1234-6789=-ef71-f3998c3e92e3/threat/email/6b5ebcb1a7fb7e19bad6d8e02a2a5e169320b1f8a75=9e7db48f3da6f4e798a39?linkOrigin=3Dnotif
violation: plugin.spec.yaml[838]: https://threatinsight.proofpoint.com/5d1ce8c6-1234-6789-ef71-f39=98c3e92e3/threat/email/6b5ebcb1a7fb7e19bad6d8e02a2a5e169320b1f8a759e7db48f3=da6f4e798a39?linkOrigin=3Dnotif\
violation: plugin.spec.yaml[838]: https://threatinsight.proofpoint.com/5d1ce8c6-1234-6789-ef71-f39=98c3e92e3/settings/privileges?linkOrigin=3Dnotif\
violation: help.md[270]: https://threatinsight.proofpoint.com/e65934ff-e650...
violation: help.md[346]: https://threatinsight.proofpoint.com/e65934ff-e650...
violation: help.md[357]: https://threatinsight.proofpoint.com/e65934ff-e650...
violation: help.md[509]: https://threatinsight.proofpoint.com/e65934ff-e650...
violation: help.md[521]: https://threatinsight.proofpoint.com/e65934ff-e650...
violation: help.md[582]: https://threatinsight.proofpoint.com/e65934ff-e650...
violation: help.md[700]: https://threatinsight.proofpoint.com/e65934ff-e650...
violation: help.md[742]: https://threatinsight.proofpoint.com/v7l34e70-a2ec-a214-bc4d-acd68a33dba2/threat/email/9c22af77f29f5eb007403455b7896906b479995b6444e421d6093e683f593e4?linkOrigin=notif
violation: help.md[135]: https://urldefense.proofpoint.com/v1/url?u=http://www.example.com/&amp
[*] Plugin failed validation! The following validation errors occurred:

Validator "HelpExampleValidator" failed! 
	Cause: Improperly formatted JSON example identified in Example input of action titled Input. Please updated the JSON example in help.md with 2 space indentation.

Validator "ConfidentialValidator" failed! 
	Cause: Please use 'user@example.com' when including emails. The above items violated this.


----
[*] Total time elapsed: 45697.529ms
```

<summary>
icon-validate --all .
</summary>
</details>

<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../../tools/Makefiles/Helpers.mk ../../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator CloudReadyValidator
[*] Executing validator UnapprovedKeywordsValidator
[*] Executing validator HelpExampleValidator
[*] Plugin failed validation! The following validation errors occurred:

Validator "HelpExampleValidator" failed! 
	Cause: Improperly formatted JSON example identified in Example input of action titled Input. Please updated the JSON example in help.md with 2 space indentation.


----
[*] Total time elapsed: 4541.5109999999995ms
```

<summary>
make validate
</summary>
</details>

### Test

Autogenerate with:
<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Proofpoint https://example.com Step name: fetch_forensics\n",
    "meta": {},
    "output": {
      "status": "Success"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/proofpoint_tap:3.1.2 --debug test < tests/fetch_forensics_campain_id.json
</summary>
</details>

### CODE REVIEW

**Review by mrinehart-r7 — APPROVED** [review]

File: plugins/proofpoint_tap/help.md [hunk 1]
```diff
@@ -975,6 +1048,7 @@ This plugin does not contain any troubleshooting information.
 
 # Version History
```
Comment (jrose-r7) [inline plugins/proofpoint_tap/help.md:1049]: Should there be a section on supported versions?

File: plugins/proofpoint_tap/plugin.spec.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ products: [insightconnect]
 name: proofpoint_tap
```
Comment (jrose-r7) [inline plugins/proofpoint_tap/plugin.spec.yaml:4]: Include a supported version section https://docs.rapid7.com/insightconnect/plugin-spec#metadata-section

**Review by jrose-r7 — DISMISSED** [review]

File: plugins/proofpoint_tap/plugin.spec.yaml [hunk 3]
```diff
@@ -4,7 +4,8 @@ products: [insightconnect]
...
 title: Proofpoint TAP
 description: Parse Proofpoint Targeted Attack Protection (TAP) alerts
-version: 3.1.1
+version: 3.1.2
+supported_versions: ["v2"]
```
Comment (jschipp-r7) [inline plugins/proofpoint_tap/plugin.spec.yaml:8]: This is a SaaS service that doesn't have versions

```suggestion
supported_versions: ["Proofpoint TAP API v2", "Tested on 2021-8-30"]
```

**Review by jschipp-r7 — APPROVED** [review]

File: plugins/proofpoint_tap/help.md [hunk 4]
```diff
@@ -115,7 +119,7 @@ Example input:
 
 ```
 {
-  "urls": ["https://example.com", "https://example2.com"]
+  "urls": "[\"https://example.com\", \"https://example2.com\"]"
```
Comment (mrinehart-r7) [inline plugins/proofpoint_tap/help.md:122]:
```suggestion
  "urls": ["https://example.com", "https://example2.com"]
```
This should also be pretty printed with newlines and tab indents

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

File: plugins/proofpoint_tap/unit_test/test_get_top_clickers.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+                            "vip": False,
+                        },
+                    }
+                ],
+            }
+        }
+        self.assertEqual(actual, expected)
+
+    @patch("requests.request", side_effect=Util.mocked_requests_get)
+    def test_get_top_clickers3(self, mock_request):
```
Comment (pmara-r7) [inline plugins/proofpoint_tap/unit_test/test_get_top_clickers.py:61]: The response on all of these is the same- why have the extra tests?

**Review by mberezin-r7 — APPROVED** [review]

**Pull request merged**